code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -194,7 +194,9 @@
 	if(.)
 		amount_grown += rand(1,2)
 		if(amount_grown >= 100)
-			new /mob/living/simple_animal/chicken(src.loc)
+			var/mob/living/simple_animal/chicken/C = new /mob/living/simple_animal/chicken(src.loc)
+			if(key)

```

Comment:
I went with how the code for turning ghosts into mobs and such works, but i can adjust this bit to use mind